doc/source/user/ogr_sql_sqlite_dialect.rst

```diff
@@ -4,8 +4,60 @@
 OGR SQL dialect and SQLITE SQL dialect
 ================================================================================
 
+The GDALDataset supports executing commands against a datasource via the
+:cpp:func:`GDALDataset::ExecuteSQL` method. How such commands are evaluated
+is dependent on the datasets.
+
+- For most file formats (e.g. Shapefiles, GeoJSON, MapInfo files), the built-in
+  :ref:`ogr_sql_dialect` dialect will be used by defaults. It is also possible
+  to request the :ref:`ogr_sql_dialect` alternate dialect to be used, which
+  will use the SQLite engine to evaluate commands on GDAL datasets.
+
+- All OGR drivers for database systems: :ref:`vector.mysql`, :ref:`vector.pg`,
+  :ref:`vector.oci`, :ref:`vector.sqlite`, :ref:`vector.gpkg`,
+  :ref:`vector.odbc`, :ref:`vector.pgeo`, :ref:`vector.hana` and :ref:`vector.mssqlspatial`,
+  override the :cpp:func:`GDALDataset::ExecuteSQL` function with dedicated implementation
+  and, by default, pass the SQL statements directly to the underlying RDBMS.
+  In these cases the SQL syntax varies in some particulars from OGR SQL.
+  Also, anything possible in SQL can then be accomplished for these particular
+  databases. Only the result of SQL WHERE statements will be returned as
```

Comment:
I know this is the old phrasing, but it's a little unclear to me. Is it supposed to say "SELECT statements"?

And it's probably not true as a general case, e.g. `delete from foo returning *` will also work on Postgres.